include/gripper/core/gripper_info.hpp

```diff
@@ -8,50 +8,38 @@
 #include "can/core/can_writer_task.hpp"
 #include "can/core/ids.hpp"
 #include "can/core/messages.hpp"
+#include "eeprom/core/serial_number.hpp"
 
 namespace gripper_info {
 using namespace can_ids;
 using namespace can_messages;
 
-struct GripperInfo {
-    uint16_t model;
-    std::array<char, 12> serial;
-};
-
 /**
  * A HandlesMessages implementing class that will respond to system messages.
  *
  * @tparam CanClient can writer task client
+ * @tparam EEPromClient eeprom task client
  */
-template <message_writer_task::TaskClient CanClient>
-class GripperInfoMessageHandler {
+template <message_writer_task::TaskClient CanClient,
+          eeprom::task::TaskClient EEPromClient>
+class GripperInfoMessageHandler : eeprom::serial_number::ReadListener {
   public:
     /**
      * Constructor
      *
      * @param writer A message writer for sending the response
+     * @param eeprom_client An eeprom task client
      */
-    explicit GripperInfoMessageHandler(CanClient &writer)
-        : GripperInfoMessageHandler(
-              writer, GripperInfo{.model = 1,
-                                  .serial = std::array{'2', '0', '2', '2', '0',
-                                                       '4', '2', '1', 'A', '0',
-                                                       '5', '\0'}}) {}
-    GripperInfoMessageHandler(CanClient &writer,
-                              const GripperInfo &gripper_info)
-        : writer(writer), response{.model = gripper_info.model} {
-        std::copy_n(
-            gripper_info.serial.cbegin(),
-            std::min(gripper_info.serial.size(), response.serial.size()),
-            response.serial.begin());
-    }
+    explicit GripperInfoMessageHandler(CanClient &writer,
+                                       EEPromClient &eeprom_client)
+        : writer{writer}, serial_number_accessor{eeprom_client, *this} {}
```

Comment:
can we just pass `this` rather than `*this` ?

Reply:
No. Why do you ask?

Reply:
` SerialNumberAccessor(EEPromTaskClient& eeprom_client,
                                  ReadListener& listener)` for some reason got me thinking we could just pass this, rather than a dereference this and pass a copy

Reply:
`this` is a pointer. We're not passing a copy.